### PR TITLE
Parse sql filter clause to create datafusion physical expression

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/src/datafusion/logical_expr.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+use arrow_schema::DataType;
+use datafusion::logical_expr::Operator;
+use datafusion::scalar::ScalarValue;
+use datafusion::{logical_expr::BinaryExpr, prelude::*};
+
+use crate::datatypes::Schema;
+use crate::{Error, Result};
+
+/// Resolve a Value
+fn resolve_value(expr: &Expr, data_type: &DataType) -> Result<Expr> {
```
Comment (changhiskhan) [inline rust/src/datafusion/logical_expr.rs:26]: needs to support list value for `item in (...)` ?

File: rust/src/io/exec/planner.rs [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+                return Err(Error::IO(format!(
+                    "Expression '{expr}' is not supported as filter in lance"
+                )))
+            }
+        }
+    }
+
+    /// Create Logical [Expr] from a SQL filter clause.
+    pub fn parse_filter(&self, filter: &str) -> Result<Expr> {
+        let sql = format!("SELECT 1 FROM t WHERE {filter}");
```
Comment (changhiskhan) [inline rust/src/io/exec/planner.rs:149]: what is this?

File: rust/src/io/exec/planner.rs [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+    fn column(&self, idents: &[Ident]) -> Result<Expr> {
+        Ok(col(idents
+            .iter()
+            .map(|id| id.value.clone())
+            .collect::<Vec<_>>()
+            .join(".")))
+    }
+
+    fn binary_op(&self, op: &BinaryOperator) -> Result<Operator> {
```
Comment (changhiskhan) [inline rust/src/io/exec/planner.rs:54]: isin?

File: rust/src/datafusion/logical_expr.rs (hunk 1 above)
Comment (eddyxu) [reply rust/src/datafusion/logical_expr.rs:26]: Ok, can do.

File: rust/src/io/exec/planner.rs (hunk 2 above)
Comment (eddyxu) [reply rust/src/io/exec/planner.rs:149]: This is trick sqlparser to parse a SQL statement, and pick the `WHERE` clause from the statement

File: rust/src/io/exec/planner.rs (hunk 3 above)
Comment (eddyxu) [reply rust/src/io/exec/planner.rs:54]: will do.

File: rust/src/io/exec/planner.rs (hunk 2 above)
Comment (changhiskhan) [reply rust/src/io/exec/planner.rs:154]: worth adding a code comment for posterity

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**